File: Data Structures & Algorithms/pow-x-n/submission-2.java
class Solution {
    public double myPow(double x, int n) {
        if (n == 0){
            return 1;
        }

        int power = 0;
        if (n > 0){
            power = n;
        } else {
            power = -n;
        }

        double num = x;
        int count = 1;

        while (count != power){
            num = num*x;
            count++;
        }
        if (n > 0){
            return num;
        }
        return 1/num;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
import heapq
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        numbers = {}
        for num in nums:
            numbers[num] = numbers.get(num, 0) + 1
        
        heap = []
        for num in numbers.keys():
            heapq.heappush(heap, [numbers[num],num])

            if len(heap) > k:
                heapq.heappop(heap)

        output = []
        for i in range(k):
            output.append(heapq.heappop(heap)[1])
        return output
